shared/reducers/chat2.js

```diff
@@ -1047,6 +1045,102 @@ const rootReducer = (
           return info.set('hits', I.List())
         }
       )
+    case Chat2Gen.setThreadSearchQuery:
+      return state.setIn(['threadSearchQueryMap', action.payload.conversationIDKey], action.payload.query)
+    case Chat2Gen.inboxSearchSetTextStatus:
+      return state.update('inboxSearch', info => {
+        return (info || Constants.makeInboxSearchInfo()).merge({
+          textStatus: action.payload.status,
+        })
+      })
+    case Chat2Gen.inboxSearchSetIndexPercent:
+      return state.update('inboxSearch', info => {
+        return (info || Constants.makeInboxSearchInfo()).merge({
+          indexPercent: action.payload.percent,
+        })
+      })
+    case Chat2Gen.toggleInboxSearch: {
+      let nextState = state
+      if (action.payload.enabled && !state.inboxSearch) {
+        nextState = state.set('inboxSearch', Constants.makeInboxSearchInfo())
+      } else if (!action.payload.enabled && state.inboxSearch) {
+        nextState = state.set('inboxSearch', null)
+      }
+      return nextState
+    }
+    case Chat2Gen.inboxSearchTextResult:
+      if (!state.inboxSearch || state.inboxSearch.textStatus !== 'inprogress') {
+        return state
+      }
+      if (!state.metaMap.get(action.payload.result.conversationIDKey)) {
+        return state
+      }
+      return state.update('inboxSearch', info => {
+        const old = info || Constants.makeInboxSearchInfo()
```

Comment:
When I do that I get these frustrating 

```
Cannot call state.update because property update is missing in object type [1].
```

Errors out of Flow which just seem like a waste of time to try and solve.

Reply:
Bummer, ok